tensorflow_io/oss/kernels/ossfs/oss_file_system.cc

```diff
@@ -0,0 +1,1158 @@
+    }
+
+    if (object[object.length() - 1] == '/') {
+      stat->is_directory = true;
+    } else {
+      stat->is_directory = false;
+    }
+
+    return Status::OK();
+  } else {
+    string msg;
+    VLOG(1) << "can not find " << object << " errMsg: " << msg;
+    oss_error_message(status, &msg);
+    return errors::NotFound("can not find ", object, " errMsg: ", msg);
+  }
+}
+
+Status OSSFileSystem::Stat(const std::string& fname, FileStatistics* stat) {
+  TF_RETURN_IF_ERROR(oss_initialize());
+  std::string object, bucket;
+  TF_RETURN_IF_ERROR(_ParseOSSURIPath(fname, &bucket, &object));
+  OSSConnection oss(host_, access_id_, access_key_);
+  oss_request_options_t* ossOptions = oss.getRequestOptions();
+  aos_pool_t* pool = oss.getPool();
+
+  return _StatInternal(pool, ossOptions, bucket, object, stat);
+}
+
+Status OSSFileSystem::GetChildren(const std::string& dir,
+                                  std::vector<std::string>* result) {
+  result->clear();
+  TF_RETURN_IF_ERROR(oss_initialize());
+  std::string object, bucket;
+  TF_RETURN_IF_ERROR(_ParseOSSURIPath(dir, &bucket, &object));
+  OSSConnection oss(host_, access_id_, access_key_);
+  oss_request_options_t* oss_options = oss.getRequestOptions();
+  aos_pool_t* pool = oss.getPool();
+  return _ListObjects(pool, oss_options, bucket, object, result, true, false,
+                      1000);
+}
+
+Status OSSFileSystem::GetMatchingPaths(const std::string& pattern,
+                                       std::vector<std::string>* results) {
+  results->clear();
+  // Find the fixed prefix by looking for the first wildcard.
+  const string& fixed_prefix =
+      pattern.substr(0, pattern.find_first_of("*?[\\"));
+  const string& dir = string(io::Dirname(fixed_prefix));
+  const string& base_prefix = string(io::Basename(fixed_prefix));
+  if (dir.empty()) {
+    return errors::InvalidArgument("A oss pattern doesn't have a bucket name: ",
+                                   pattern);
+  }
+
+  std::string new_uri = dir;
+  size_t params_pos = pattern.find_last_of('?');
+  if (params_pos != std::string::npos) {
+    new_uri.append(pattern.substr(params_pos));
+  }
+
+  TF_RETURN_IF_ERROR(oss_initialize());
+  std::string object, bucket;
+  TF_RETURN_IF_ERROR(_ParseOSSURIPath(new_uri, &bucket, &object));
+
+  OSSConnection oss(host_, access_id_, access_key_);
+  oss_request_options_t* oss_options = oss.getRequestOptions();
+  aos_pool_t* pool = oss.getPool();
+
+  std::vector<std::string> listing;
+  TF_RETURN_IF_ERROR(_ListObjects(pool, oss_options, bucket, object, &listing,
+                                  true, false, 1000));
+
+  // Match all obtained paths to the input pattern.
+  for (auto path : listing) {
+    if (str_util::StartsWith(StringPiece(path), base_prefix)) {
+      const string file_path = io::JoinPath(dir, path);
+      if (Env::Default()->MatchPath(file_path, pattern)) {
+        results->push_back(dir + "/" + path);
```

Comment:
Perhaps move "/" to a constant? Also there's inconsistent use of single and double quotes.

Reply:
What inconsistency are you talking about? In C/C++ single and doule qutoes denote different types.

Reply:
Think I misread it.